lib/services/BookmarkService.dart

```diff
@@ -22,27 +17,39 @@ class BookmarkService extends Service {
   BookmarkService._(user) : super(endpointBase: _ENDPOINT_BASE, user: user);
 
   Future<List<Bookmark>> retrieveAllBookmarks() async {
-    String basicAuth = 'Basic ' + base64Encode(utf8.encode('${this.user.username}:${this.user.appPassword}'));
-    http.Response response = await getRequest();
+    Map<String, dynamic> response = await getRequest('?page=-1');
 
     List<Bookmark> result = List<Bookmark>();
-    if (response.statusCode == 200 && isJSON(response.body)) {
-      Map<String, dynamic> parsed = jsonDecode(response.body);
-      if (parsed["status"] != "success") return result;
+    if (response["status"] != "success") return result;
 
-      for (Map<String, dynamic> element in parsed["data"]) {
-        result.add(Bookmark.fromJson(element));
-      }
+    for (Map<String, dynamic> element in response["data"]) {
+      result.add(Bookmark.fromJson(element));
     }
 
     return result;
   }
-  
+
+  Future<List<Bookmark>> retrieveBookmarksOfFolder(int folderId) async {
+    Map<String, dynamic> response =
+        await getRequest('?page=-1&folder=' + folderId.toString());
+
+    List<Bookmark> result = List<Bookmark>();
+    if (response["status"] != "success") return result;
+
+    for (Map<String, dynamic> element in response["data"]) {
+      Bookmark bookmark = Bookmark.fromJson(element);
+      bookmark.setFavicon(getImage('/' + bookmark.id.toString() + '/favicon'));
+      result.add(bookmark);
+    }
+
+    return result;
+  }
+
   void delete(Bookmark bookmark) {
-    super.deleteRequest(bookmark.id);
+    super.deleteRequest('/' + bookmark.id.toString());
```

Comment:
Could you move the `/` prefixing into the super class?

Reply:
What about `getRequest('?page=-1')`, though?

Reply:
True. That edge case needs handling. Something like this maybe?
```
if (path.at(0) == '/' || path.at(0) == '?') return doRequest(path);

return doRequest(prependSlash(path));
```

Reply:
Done :)